src/lib.rs

```diff
@@ -46,6 +46,17 @@ pub struct Matrix22 {
     pub a22: BigInt,
 }
 
+impl Matrix22 {
+    pub fn mul(&self, another: Self) -> Self {
```

Comment:
I could be a good idea to only borrow `another` with `anther: &Self`, in case it needs to be used more that once